😈 1.8.1 (#1116)
cloudflare/wrangler: CHANGELOG.md

# Changelog

## 🙈1.8.1

- ### Fixes

  - **Fix live reload for `wrangler dev` - [EverlastingBugstopper], [issue/1082][pull/1117]**

>     Updates to a Worker script in a text editor would not trigger the terminal running `wrangler dev` to update automatically. This fix makes sure the multiple threads running on wrangler get an updated lock whenever a reload is detected instead of depending on routine lock checks.

Reviewer: ```suggestion
    `wrangler dev` re-builds and re-uploads your script to the Cloudflare API when it detects a file change. The Cloudflare API returns a new token which allows `wrangler dev` to route subsequent requests to the new script. Previously, `wrangler dev` would re-build, re-upload, and receive the new token, but it wouldn't use it for a couple of minutes due to some faulty threading logic. (darn mutexes!) After this change, `wrangler dev` will block incoming requests when it is switching the token, thus fixing the issue.
```

>     Updates to a Worker script in a text editor would not trigger the terminal running `wrangler dev` to update automatically. This fix makes sure the multiple threads running on wrangler get an updated lock whenever a reload is detected instead of depending on routine lock checks.

Reviewer: ```suggestion
  - **Fix live reload for `wrangler dev` - [EverlastingBugstopper], [issue/1082][pull/1117]**

    `wrangler dev` re-builds and re-uploads your script to the Cloudflare API when it detects a file change. The Cloudflare API returns a new token which allows `wrangler dev` to route subsequent requests to the new script. Previously, `wrangler dev` would re-build, re-upload, and receive the new token, but it wouldn't use it for a couple of minutes due to some faulty threading logic. (darn mutexes!) After this change, `wrangler dev` will block incoming requests when it is switching the token, thus fixing the issue.
```

Reviewer: Body should be updated as well here

- ### Maintenance

  - **Error messaging for internet required to talk to Cloudflare API - [EverlastingBugstopper], [issue/1093][pull/1114]**

    With the release of `wrangler dev` in 1.8.0, it was not clear to users that internet is required since the feature communicates with Cloudflare's API. With this error message, users without internet connection are shown actionable next steps - check internet connection and lastly check if Cloudflare's API is down.

  - **Remove unneeded carriage return in `wrangler config` - [gabbifish ], [issue/1009][pull/1112]**

>     Removes the unneed carriage return from `wrangler config`.

Reviewer: we need more context here

>     Previously interactive input from `wrangler config` unexpectedly added a carriage return. This fix removes the unneed carriage return from the input collected.

Reviewer: ```suggestion
    Previously, interactive input from `wrangler secret put` added a carriage return to the secret key/value pairs on Windows. This no longer happens and input is parsed properly before uploading.
```
